casper/src/main/scala/coop/rchain/casper/MultiParentCasperConstructor.scala

```diff
@@ -95,8 +95,10 @@ sealed abstract class MultiParentCasperConstructorInstances {
             "No valid ApprovedBlock has been received to instantiate a Casper instance!")))(
           _.toEither)
 
-      override def lastApprovedBlock: F[Option[ApprovedBlock]] =
-        genesisPromise.future.value.flatMap(_.toOption.map(_._1)).pure[F]
+      override def lastApprovedBlock: F[Option[ApprovedBlock]] = Capture[F].capture {
```

Comment:
Can we use `Sync[F].capture { … }` ? We have to add `Sync` anyway because all the `cats`' goodies require that and having both `Capture` and `Sync` is a burden. We should move to `Sync` some day.